.github/workflows/deploy.yaml

```
@@ -30,15 +30,50 @@ jobs:
         role-session-name: GitHub_to_AWS_via_FederatedOIDC
         aws-region: ${{ env.AWS_REGION }}
 
+    - name: Check CloudFormation stack existence and status
+      id: check_stack
+      run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:5:34: Double quote to prevent globbing and word splitting

(shellcheck)